### PR TITLE
Add instructions for Go custom container builds, link from 2.55.0 release notes.

### PR DESCRIPTION
**Pull request by lostluck**
Add a message to 2.55.0's Breaking Changes regarding the Go container moving to distroless. 

While the existing instructions for
https://beam.apache.org/documentation/runtime/environments/#modify-existing-base-image do work, the simplicity of Go's handling in this case is valuable to emphasize.

Staging:
https://apache-beam-website-pull-requests.storage.googleapis.com/30290/documentation/runtime/environments/index.html#from-scratch-go

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @liferoad

File: CHANGES.md [hunk 1]
```diff
@@ -72,6 +72,10 @@
 
 * X behavior was changed ([#X](https://github.com/apache/beam/issues/X)).
 * Arrow version was bumped to 15.0.0 from 5.0.0 ([#30181](https://github.com/apache/beam/pull/30181)).
+* Go SDK users who build custom worker containers may run into issues with the move to distroless containers as a base (see Security Fixes).
+  * The recommendation is to confinue to use the container for 2.53.0 as a base for now eg. `apache/beam_go_sdk:2.53.0`.
```
Comment (liferoad) [inline CHANGES.md:76]: do you mean 2.54.0 and earlier versions?

Comment (lostluck) [reply CHANGES.md]: Whoops. Thank you.

**Review by liferoad — APPROVED** [review]

**Comment by lostluck** [comment]
I got into a groove and went and figured out the right custom container instructions, so no stopgap required.
Did you know Beam works just fine on Alpine when the container is set up properly? (As an example.)

**Comment by liferoad** [comment]
> I got into a groove and went and figured out the right custom container instructions, so no stopgap required. Did you know Beam works just fine on Alpine when the container is set up properly? (As an example.)

I never try that but it should work. 

BTW: we can also add https://cloud.google.com/dataflow/docs/guides/using-custom-containers to the md file if users need more information about custom containers.

**Comment by lostluck** [comment]
> BTW: we can also add https://cloud.google.com/dataflow/docs/guides/using-custom-containers to the md file if users need more information about custom containers.

That file just links to the page I'm already linking to. It's also weird to link to the Dataflow pages from a Beam release for a non-dataflow feature.

**Comment by liferoad** [comment]
I mean adding this to website/www/site/content/en/documentation/runtime/environments.md if users want to learn more how to do this in general. But it is fine not to do this if you feel it is strange. :)

**Comment by lostluck** [comment]
I don't feel it's reasonable to conflate adding that link to this PR. I have no idea where I'd put it.

I've always found our strategy of essentially putting general Beam content on the Dataflow pages to be baffling, vs improving the content on the Beam pages.

**Pull request merged**